libs/route-config/src/lib/route-config.service.ts

```diff
@@ -19,40 +20,63 @@ export type RouteData<
 
 export type RouteDataParam<ConfigParamsNames extends string> = keyof RouteData<ConfigParamsNames>;
 
+const gatherRoutes = (activatedRoute: ActivatedRoute): ActivatedRoute[] => {
+  const routes: ActivatedRoute[] = activatedRoute.pathFromRoot;
+
+  let route = activatedRoute.firstChild;
+  while (route) {
+    routes.push(route);
+    route = route.firstChild;
+  }
+
+  return routes;
+};
+
 @Injectable()
 export class RouteConfigService<
   RouteTags extends string = string,
   ConfigParamsNames extends string = never
 > {
-  constructor(private activatedRoute: ActivatedRoute, private router: Router) {}
+  private get injectedDefaultValue(): Partial<RouteData<ConfigParamsNames, RouteTags>> {
+    return this._injectedDefaultValue || {};
+  }
 
-  getLeafConfig(paramName: 'routeTags', defaultValue: RouteTags[]): Observable<RouteTags[]>;
-  getLeafConfig<T>(paramName: ConfigParamsNames, defaultValue: T): Observable<T>;
-  getLeafConfig<T = unknown>(
-    paramName: RouteDataParam<ConfigParamsNames>,
-    defaultValue: T
-  ): Observable<T> {
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private router: Router,
+    @Optional()
+    @Inject(ROUTE_DATA_DEFAULT_VALUE)
+    private _injectedDefaultValue?: Partial<RouteData<ConfigParamsNames, RouteTags>>
+  ) {}
+
+  getWholeLeafConfig<
```

Comment:
🤔  shouldn't this be like `getActivatedRouteConfig`?

Reply:
done